Security/OAuthUtils.php

```diff
@@ -117,6 +117,18 @@ public function getAuthorizationUrl(Request $request, $name, $redirectUrl = null
             }
         }
 
+        // grab extra parameters from session
+
+        if ($request->hasSession()) {
+            // initialize the session for preventing SessionUnavailableException
+            $session = $request->getSession();
+            $session->start();
+            $sessionKey = "_security.$name.parameters";
+            $extraSessionParameters = $session->get($sessionKey, []);
+            $extraParameters = array_merge($extraParameters, $extraSessionParameters);
+            $session->remove($sessionKey);
+        }
+
```

Comment:
Fixes issue: Missing refresh tokens after signin up with permissions granted.

This allows to perform an auth retry with additional options, after an error is detected.